lib/backend.js

```diff
@@ -40,6 +41,8 @@ function Backend(options) {
   this.agentsCount = 0;
   this.remoteAgentsCount = 0;
 
+  logger.setMethods(options.logger);
```

Comment:
Now that I've read through the code, I do see what you mentioned around module caching - The `options.logger` is passed into the Backend constructor on instance creation but is then set onto the logger module, which is a process-level singleton. It is indeed a bit odd to call a constructor and have it change some module-level configuration.

If we don't want to pass around logger instances everywhere, perhaps a compromise would be to have the server also use a static config call, like you're doing with the client. And a name like `ShareDB.setDefaultLogger({...})` on server and client could provide a hint that it's not an instance-level logger.

Reply:
@ericyhwang I've pushed a change that moves this to static configuration just like the client:

```javascript
Backend.logger.setMethods({...});
```

I personally think that this static accessor should make it clear to consumers that this will affect _all_ instances of `Backend`.

If you still want to change the method name, I'm not a huge fan of `setDefaultLogger`, because it implies there's some way of setting it on an instance-by-instance (non-default) way (which there isn't). I'd possibly prefer something along the lines of `setGlobalLoggingMethods` or similar.

Thoughts?

Reply:
How about `setLogger`?

Reply:
I kind of think `logger.setLogger` is a bit weird, but I don't feel strongly.

Reply:
Maybe..? But this is sort of why I landed on `setMethods`, because I thought it was nicer to be explicit about what it is you're setting.

Reply:
I think `setMethods` is good, thanks!